packages/cli-kit/src/ui.ts

```diff
@@ -48,6 +48,31 @@ interface BaseQuestion<TName extends string> {
   result?: (value: string) => string | boolean
 }
 
+export interface SimpleTaskOptions {
+  title: string
+  success?: string
+  task: () => Promise<void | string>
+}
+
+/**
+ * Performs a task with the title kept up to date and stdout available to the
+ * task while it runs (there is no re-writing stdout while the task runs).
+ */
+export const simpleTask = async ({title, success: _success, task}: SimpleTaskOptions) => {
+  let success
+  started(title, logUpdate)
+  try {
+    const result = await task()
+    success = _success || result || title
+  } catch (err) {
+    failed(title, logUpdate)
+    logUpdate.done()
+    throw err
+  }
+  completed(success, logUpdate)
```

Comment:
`logUpdate` has a `logUpdateStderr` interface to log through `stderr` instead of `stdout`. Should we use that one here?

Reply:
We want to re-write the task's original `title` line indicating that the task succeeded or failed. The original title line is written by a logUpdate using stdout, so we want to use that same created logUpdate to re-write it. If we use a new logUpdate created from the logUpdateStderr, it would not rewrite the title line, since this state is [kept in a closure for each logUpdate](https://github.com/sindresorhus/log-update/blob/main/index.js#L36-L38). Throwing the error allows the CLI to write the details of the error to stderr after re-writing the title to indicate the failure.